docs/asciidoc/http-client.adoc

```diff
@@ -367,3 +368,69 @@ include::{examplesdir}/uds/Application.java[lines=18..33]
 ----
 <1> Specifies `DomainSocketAddress` that will be used
 ====
+== Host Name Resolution
+By default, the `HTTP` client uses Netty's domain name lookup mechanism that resolves a domain name asynchronously.
```

Comment:
we could use the `HttpClient` class name here instead of `HTTP client`. class name is descriptive and more relevant to be marked as code than just the protocol, wdyt?